server.js

```diff
@@ -124,8 +109,36 @@ app.use('/api', function (req, res, next) {
 
 app.use('/api', require('./server/routes'))
 
-app.get('*', function (req, res) {
-  res.sendFile(path.resolve(__dirname, 'client/build', 'index.html'))
+app.get('*', function (req, res, next) {
+  var file = path.resolve(__dirname, 'client/build', 'index.html')
+  fs.readFile(file, function (err, data) {
+    if (err) return next(err)
+
+    const clientPkg = require('./client/package.json')
+    const rumConfig = {
+      serviceName: process.env.ELASTIC_APM_JS_SERVICE_NAME || clientPkg.name,
+      serviceVersion: process.env.ELASTIC_APM_JS_SERVICE_VERSION || clientPkg.version,
+      serverUrl: process.env.ELASTIC_APM_JS_SERVER_URL || 'http://localhost:8200',
+      pageLoadTraceId: undefined,
+      pageLoadSpanId: undefined,
+      pageLoadSampled: undefined
+    }
+
+    const transaction = apm.currentTransaction
+    if (transaction) {
+      rumConfig.pageLoadTraceId = transaction.traceId
+      rumConfig.pageLoadSpanId = transaction.ensureParentId()
+      rumConfig.pageLoadSampled = transaction.sampled
+    }
+
+    const body = data.toString()
+      .replace('<script type="text/javascript" src="/rum-config.js"></script>', '')
+      .replace('<head>', `<head><script>window.rumConfig = ${JSON.stringify(rumConfig)}</script>`)
```

Comment:
@jahtalab do the RUM agent support parsing a JSON string like here in `rumConfig` or should it be a proper JavaScript object?

Reply:
Ruby doesn `to_json`, so it's the same there. https://github.com/elastic/opbeans-ruby/commit/72676346748083292998b7b29bd2230cf246fc9b#diff-55c5b7aecfb519d0e4880eaf2788eb6eR21

Reply:
@watson , This is not a RUM agent feature it's only what I have implemented in opbeans-frontend. We don't parse JSON string (you can see the code [here](https://github.com/elastic/opbeans-frontend/blob/master/src/rum.js#L43)) so currently `rumConfig` should be an object on the window.